velox/functions/sparksql/tests/AtLeastNNonNullsTest.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/specialforms/AtLeastNNonNulls.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+using namespace facebook::velox::test;
+
+static constexpr auto kNaNDouble = std::numeric_limits<double>::quiet_NaN();
+static constexpr auto kNaNFloat = std::numeric_limits<float>::quiet_NaN();
+static constexpr auto kMaxDouble = std::numeric_limits<double>::max();
+static constexpr auto kMaxFloat = std::numeric_limits<float>::max();
+
+class AtLeastNNonNullsTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  core::CallTypedExprPtr createAtLeastNNonNulls(
```

Comment:
Why can't we use a parser and provide SQL string?

```
at_least_n_non_nulls(2, a, b, c)
```

Reply:
I just reference the impl of `DecimalRoundTest`, yes, above is simplier, I changed to it. Thanks.

Reply:
Would be nice to refactor DecimalRoundTest in a separate PR.